libsolidity/analysis/TypeChecker.cpp

```diff
@@ -480,6 +480,10 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 				"Initial value for constant variable has to be compile-time constant."
 			);
 	}
+	else if (_variable.immutable())
+		if (_variable.type()->isValueType())
```

Comment:
```suggestion
	else if (_variable.immutable() && _variable.type()->isValueType())
```

Reply:
I did it like this to keep the symmetry with the `isConstant()` check above.